statsmodels/discrete/discrete_model.py

```diff
@@ -2371,11 +2371,10 @@ def __init__(self, model, mlefit, cov_type='nonrobust', cov_kwds=None,
 
 
     def __getstate__(self):
-        try:
-            #remove unpicklable callback
-            self.mle_settings['callback'] = None
-        except (AttributeError, KeyError):
-            pass
+        # remove unpicklable methods
+        self.mle_settings['callback'] = None
```

Comment:
I don't remember if there was a case where it failed.  
Maybe it was just risk aversion to imposing details that are not important or required parts of the general interface, i.e. defensive programming. Or it was written in analogy to a `base` function.
Or just generic future proofness, in case we add estimators that don't define mle_settings and a callback key.